packages/components/app/styles/components/dropdown.scss

```diff
@@ -408,8 +416,86 @@ $hds-dropdown-toggle-border-radius: 5px;
   }
 }
 
+// HDS::DROPDOWN::LIST-ITEM::TITLE
+
 .hds-dropdown-list-item--title {
   padding: 10px 16px 4px;
   color: var(--token-color-foreground-strong);
 }
 
+// HDS::DROPDOWN::LIST-ITEM::CHECKMARK
+
+.hds-dropdown-list-item--checkmark {
+  a.hds-dropdown-list-item__interactive,
+  button.hds-dropdown-list-item__interactive {
+    // leave room for the checkmark to align the result count on non-selected items
+    padding-right: 25px; // 9px default + 16px for the 'check' FlightIcon
+
+    // disable background on focus and active states
+    &:focus,
+    &:active,
+    &.mock-focus,
+    &.mock-active {
+      color: var(--token-color-foreground-action-active);
+
+      &::after {
+        background-color: inherit;
+      }
+    }
+  }
+
+  .hds-dropdown-list-item__interactive-text {
+    flex: 1;
+  }
+}
+
+.hds-dropdown-list-item--checkmark-selected {
+  a.hds-dropdown-list-item__interactive,
+  button.hds-dropdown-list-item__interactive {
+    padding-right: 9px; // reset to default
+    color: var(--token-color-foreground-action);
+  }
+}
+
+.hds-dropdown-list-item__checkmark {
+  align-self: flex-start;
+  margin-top: 0.125rem;
+}
+
+// HDS::DROPDOWN::LIST-ITEM::CHECKBOX & HDS::DROPDOWN::LIST-ITEM::RADIO
+
+.hds-dropdown-list-item--checkbox,
+.hds-dropdown-list-item--radio {
+  display: flex;
+  padding: 8px 16px;
+
+  .hds-dropdown-list-item__control {
+    flex-shrink: 0;
+    margin-top: 2px;
+    margin-right: 8px;
+  }
+
+  .hds-dropdown-list-item__label {
+    flex-grow: 1;
+    width: fit-content;
+    color: var(--token-color-foreground-primary);
+  }
+}
+
+// RESULT COUNT
+.hds-dropdown-list-item__count {
+  align-self: flex-start;
+  padding: 0.125rem 0.5rem;
+  color: var(--token-color-foreground-faint);
+  line-height: 1.2308; // 16px
```

Comment:
maybe instead of doing this double trick of setting a padding + a line height, you can be more declarative and set the line-height to `20px` (the line-height of the main text) and no vertical padding (adding a comment explaining why this 20px value)

Reply:
that makes sense, especially now when the checkmark icon also uses this value – updated in [3991ef2](https://github.com/hashicorp/design-system/pull/1185/commits/3991ef22579e58c852e1f7cbadc57b66b81487a9)